jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -110,26 +145,54 @@ void load_and_test_method(
   remove_virtual_functions(model_function);
 
   // Then test both situations.
-  THEN(
-    "Code should work when remove returns is called before "
-    "replace_java_nondet.")
+  THEN("Replace nondet should work when remove returns has been called.")
   {
     remove_returns(model_function, [](const irep_idt &) { return false; });
 
     replace_java_nondet(model_function);
 
-    validate_method_removal(goto_function.body.instructions);
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  THEN("Replace nondet should work when remove returns hasn't been called.")
+  {
+    replace_java_nondet(model_function);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
```

Comment:
The `THEN` says when remove_returns hasn't been called, but you're calling it?

Reply:
`remove_returns` hasn't been called when `replace_java_nondet` is called.